### DISCUSSION TITLE
Smart mask does not bite on spider web - constant motion detected

### DISCUSSION THREAD

**Discussion by leppis** [Motion-Project/motion#1476] [General]
0. Reviewed guide and contributing documents? (Yes/No): No
1. version [x.y.z, hash, other]: 4.1.1
2. installed as a package or compiled from sources [deb, rpm, git, other]: git
3. standalone or part of third party [motion, MotionEyeOS, other]: motion
4. video stream source [V4L (card or USB), net cam (mjpeg, rtsp, other), mmal]: rtsp, Camera model SunEyes SP-P901W (1280x960)
5. hardware [x86, ARM, other]: ARM (Raspberry Pi 2)
6. operating system [16.04, Stretch, etc, FreeBSD, other]: Raspbian Wheezy

I'm having a spider problem with smart mask. Spider web is a sharp and bright object that is located in front of the infrared led, because all the bugs are flying in front of the same led. The spider web is shown on the camera view and the web keeps moving in the wind in such a way that it triggers the motion detection from t=0. See upper left corner from my video https://www.youtube.com/watch?v=60kmXci16Pw

The smart mask seems to be implemented in way that when motion is being detected, the smart mask does not build up. Now the constant motion of spider web results in a motion being detected forever (until the web is cleaned off). Smart mask does not have a chance to ignore the spider web.

I've already implemented a quick and dirty fix by allowing the smart mask to build up when motion is being detected, and it seems to bite the moving web problem. But I'm worried about a moving fog or snow, where smart mask would build up filling 100% of the image and it would stop recording. A situation where I would like to record the video anyway, since motion detection is impossible.

Any thoughts on how to solve the problem?

**Comment by denzen84** [comment]
I have the same problem with the rain in front camera in IR-light. See sample video [here](http://flangeneer.sknt.ru/ivc-kolpak/101-703-20200521-032603.mp4)

**Comment by klaernie** [comment]
Maybe a way to work around this would be to externalize the IR LEDs and disable the camera built-in LEDs. This would prevent critters from getting drawn to the lens (the kind that likes IR light) and would also mean the the rain or spider web is not directly illuminated from the camera's side.

I found some pretty cheap IR emitters online, that cost not more that 15 bucks, but light an entire room in IR.
